lighthouse-core/test/audits/service-worker-test.js

```diff
@@ -5,39 +5,167 @@
+      status: sw.status,
+      scriptURL: sw.scriptURL,
+    });
+
+    const scopeURL = sw.scopeURL || getBaseDirectory(sw.scriptURL);
+    assert.ok(scopeURL.endsWith('/')); // required by SW spec.
+
+    artifact.registrations.push({
+      registrationId: `${index}`,
+      scopeURL,
+    });
+  });
+
+  return {
+    ServiceWorker: artifact,
+    URL: {finalUrl},
+  };
+}
+
 describe('Offline: service worker audit', () => {
   it('passes when given a matching service worker version', () => {
-    const output = Audit.audit({
-      ServiceWorker: {
-        versions: [{
-          status: 'activated',
-          scriptURL: 'https://example.com/sw.js',
-        }],
-      },
-      URL: {finalUrl: 'https://example.com'},
-    });
+    const finalUrl = 'https://example.com';
+    const swOpts = [{
+      status: 'activated',
+      scriptURL: 'https://example.com/sw.js',
+    }];
 
+    const output = ServiceWorker.audit(createArtifacts(swOpts, finalUrl));
     assert.equal(output.rawValue, true);
   });
 
+  it('fails when matching service worker is not activated', () => {
+    const finalUrl = 'https://example.com';
+    const swOpts = [{
+      status: 'redundant',
+      scriptURL: 'https://example.com/sw.js',
+    }];
+
+    const output = ServiceWorker.audit(createArtifacts(swOpts, finalUrl));
+    assert.equal(output.rawValue, false);
+  });
+
   it('discards service worker registrations for other origins', () => {
-    const versions = [{
+    const finalUrl = 'https://example.com';
+    const swOpts = [{
       status: 'activated',
       scriptURL: 'https://other-example.com',
     }];
 
-    const output = Audit.audit({
-      ServiceWorker: {
-        versions,
-      },
-      URL: {finalUrl: 'https://example.com'},
-    });
+    const output = ServiceWorker.audit(createArtifacts(swOpts, finalUrl));
+    assert.equal(output.rawValue, false);
+  });
+
+  it('fails when URL is out of scope', () => {
+    const finalUrl = 'https://example.com/index.html';
+    const swOpts = [{
+      status: 'activated',
+      scriptURL: 'https://example.com/serviceworker/sw.js',
+    }];
+
+    const output = ServiceWorker.audit(createArtifacts(swOpts, finalUrl));
+    assert.equal(output.rawValue, false);
+  });
+
+  it('fails when explicit scopeURL puts the URL out of scope', () => {
+    const finalUrl = 'https://example.com/index.html';
+    const swOpts = [{
+      status: 'activated',
+      scriptURL: 'https://example.com/sw.js',
+      scopeURL: 'https://example.com/serviceworker/',
+    }];
+
+    const output = ServiceWorker.audit(createArtifacts(swOpts, finalUrl));
+    assert.equal(output.rawValue, false);
+  });
+
+  it('passes when outside default scope but explicit scopeURL puts it back in', () => {
+    const finalUrl = 'https://example.com/index.html';
+    const swOpts = [{
+      status: 'activated',
+      scriptURL: 'https://example.com/serviceworker/sw.js',
+      scopeURL: 'https://example.com/',
```

Comment:
No, good point. I went too wild with the possible combinations of states :)

Reply:
OK, actually I was wrong again. The service worker's directory is the "max scope" and it can only be narrowed by `serviceWorker.register()`, however if you serve the SW with a `Service-Worker-Allowed` header, you can change the max scope to something higher up. Google Maps does this, for instance.

Reply:
added a comment to explain

Reply:
ah OK, they should update some of those docs :)

https://www.w3.org/TR/service-workers-1/#service-worker-script-response was a pretty good example too if ya want to throw that example in there